src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -54,21 +47,7 @@ private JsonTokenType TokenType
         /// <exception cref="ObjectDisposedException">
         ///   The parent <see cref="JsonDocument"/> has been disposed.
         /// </exception>
-        public JsonValueKind ValueKind
-        {
-            get
-            {
-                if (IsImmutable)
```

Comment:
Remove the `IsImmutable` property as well (and from the ref)? It will never be true without `JsonNode` or a writable DOM equivalent.